juspay/hyperswitch
# refactor(connector): [Worldline][Worldpay][Zen] Mask PII data

## Type of Change



## Description

Mask pii information passed and received in the connector request and response for Worldline, Worldpay and zen.


## Test Case

_Note: Zen Googlepay and Worldpay can't be tested in sandbox_

1. Create a card payment with worldpay

2. Check if all the sensitive data in the `masked_response` is masked

```
curl --location '{{base_url}}/analytics/v1/connector_event_logs?type=Payment&payment_id={{payment_id}}' \
--header 'sec-ch-ua: "Not A(Brand";v="99", "Google Chrome";v="121", "Chromium";v="121"' \
--header 'sec-ch-ua-mobile: ?0' \
--header 'authorization: Bearer JWT_token' \
--header 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36' \
--header 'Content-Type: application/json' \
--header 'Referer: https://integ.hyperswitch.io/' \
--header 'api-key: {{api-key}}' \
--header 'sec-ch-ua-platform: "macOS"'
```

Response 
```
request\":\"{\\\"cardPaymentMethodSpecificInput\\\":{\\\"card\\\":{\\\"cardNumber\\\":\\\"456735**********\\\",\\\"cardholderName\\\":\\\"*** alloc::string::String ***\\\",\\\"cvv\\\":\\\"*** alloc::string::String ***\\\",\\\"expiryDate\\\":\\\"*** alloc::string::String ***\\\"},\\\"requiresApproval\\\":false,\\\"paymentProductId\\\":1},\\\"order\\\":{\\\"amountOfMoney\\\":{\\\"amount\\\":20000,\\\"currencyCode\\\":\\\"PLN\\\"},\\\"customer\\\":{\\\"billingAddress\\\":{\\\"city\\\":\\\"San Fransico\\\",\\\"countryCode\\\":\\\"US\\\",\\\"houseNumber\\\":null,\\\"state\\\":\\\"*** alloc::string::String ***\\\",\\\"stateCode\\\":null,\\\"street\\\":null,\\\"zip\\\":\\\"*** alloc::string::String ***\\\"},\\\"contactDetails\\\":{\\\"emailAddress\\\":\\\"*********@gmail.com\\\",\\\"mobilePhoneNumber\\\":null}},\\\"references\\\":{\\\"merchantReference\\\":\\\"pay_Cm4XoGAkTeXdAaAx3OUo_1\\\"}},\\\"shipping\\\":{\\\"city\\\":\\\"San Fransico\\\",\\\"countryCode\\\":\\\"US\\\",\\\"houseNumber\\\":null,\\\"name\\\":{\\\"firstName\\\":\\\"*** alloc::string::String ***\\\",\\\"surname\\\":\\\"*** alloc::string::String ***\\\",\\\"surnamePrefix\\\":null,\\\"title\\\":null},\\\"state\\\":\\\"*** alloc::string::String ***\\\",\\\"stateCode\\\":null,\\\"street\\\":null,\\\"zip\\\":\\\"*** alloc::string::String ***\\\"}}\",\"masked_response\":\"{\\\"payment\\\":{\\\"id\\\":\\\"000000113200000139460000100001\\\",\\\"status\\\":\\\"CAPTURE_REQUESTED\\\",\\\"capture_method\\\":\\\"automatic\\\"},
```

## Checklist
